extensions/grpc/codegen/src/main/java/io/quarkus/grpc/deployment/GrpcCodeGen.java

```diff
@@ -354,6 +377,22 @@ private String osClassifier() throws CodeGenException {
         }
     }
 
+    private static Predicate<Path> createPathMatcherPredicate(List<String> patterns, boolean defaultIfEmpty) {
+        if (patterns.isEmpty())
+            return (Path p) -> defaultIfEmpty;
+
+        FileSystem fs = FileSystems.getDefault();
+        List<PathMatcher> pathMatcherPredicates = patterns.stream().map(g -> fs.getPathMatcher("glob:" + g))
+                .collect(Collectors.toList());
+
+        return new Predicate<Path>() {
+            @Override
+            public boolean test(Path path) {
+                return pathMatcherPredicates.stream().anyMatch(pathMatcherPredicate -> pathMatcherPredicate.matches(path));
```

Comment:
Why having 2 streams? Might be better/simpler to do that with a single one:

```
patterns.stream().map(g -> fs.getPathMatcher("glob:" + g)
    .anyMatch(pathMatcherPredicate -> pathMatcherPredicate.matches(path));
```

Also, we may want to avoid lambdas here.

Reply:
> Why having 2 streams? Might be better/simpler to do that with a single one:
> 
> ```
> patterns.stream().map(g -> fs.getPathMatcher("glob:" + g)
>     .anyMatch(pathMatcherPredicate -> pathMatcherPredicate.matches(path));
> ```
> 
> Also, we may want to avoid lambdas here.

To avoid recreating a PathMatcher for each file. Looking at the code, it involves parsing the type (regex or glob) and compile the expression.

Will add documentation and remove lambda